src/gfx/pal_packing.cpp

```diff
@@ -199,21 +200,35 @@ class AssignedProtos {
 		return colors.size() <= options.maxOpaqueColors();
 	}
 
+	// The formula we want to compute[1] is:
+	//     sum((1 / <how many of our proto-pals contain `color`>)  for `color` in `protoPal`)
+	// But, this yields a non-integer result. Using floating-point numbers would lead
+	// to imprecision, and even per-platform differences. Not good.
+	// We avoid this by multiplying the formula such that the division always produces an integer;
+	// the LCM of all values the denominator can take is the smallest suitable factor.
+	//
+	// [1]: The paper and the associated code disagree on the formula: the code adds 1 to the
+	// denominator, whereas the paper does not; its lack causes a division by 0 if the color is not
+	// found in any proto-pal, and adding it does not seem to invalidate the paper's reasoning,
+	// so it seems safe to assume the paper is wrong.
```

Comment:
```suggestion
	// The `relSizeOf` method below should compute the sum for each color in `protoPal` of
	// the reciprocal of the "multiplicity" of the color in this collection of proto-palettes.
	// However, literally computing the reciprocals would involve floating-point division, which
	// leads to imprecision and even platform-specific differences.
	// We avoid this by multiplying the reciprocals by a factor such that division always produces
	// an integer; the LCM of all values the denominator can take is the smallest suitable factor.
```

Reply:
This rephrasing matches the paper: "the **relative size** of *t* on *p* is the sum of the reciprocals of the multiplicities of the symbols of *t in *p*", and "the **multiplicity** *µ*_*p*(*α*) counts the occurrences of *α* in the tiles of *p*".